lib/Mojo/Base.pm

```diff
@@ -101,6 +105,16 @@ sub tap {
   return $self;
 }
 
+sub with_roles {
+  if ($_[0]->can_roles()) {
+    Carp::croak "class and roles must be package names" if !!grep { ref $_ } @_;
```

Comment:
create_class_with_roles will error if anything other than a valid role name is passed, so probably not needed